hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/JoinRequestOp.java

```diff
@@ -38,9 +42,24 @@ public JoinRequestOp(JoinRequest request) {
     @Override
     public void run() {
         ClusterServiceImpl cm = getService();
+        preparePreOp(request.getPreJoinOperation());
         cm.getClusterJoinManager().handleJoinRequest(request, getConnection());
     }
 
+    private void preparePreOp(OnJoinOp preOp) {
+        if (preOp == null) {
+            return;
+        }
+
+        ClusterServiceImpl clusterService = getService();
+        NodeEngineImpl nodeEngine = clusterService.getNodeEngine();
+
+        preOp.setNodeEngine(nodeEngine);
+        OperationAccessor.setCallerAddress(preOp, getCallerAddress());
+        OperationAccessor.setConnection(preOp, getConnection());
+        preOp.setOperationResponseHandler(createEmptyResponseHandler());
+    }
```

Comment:
Is this required? There is also `OnJoinOp.beforeRun()`.

Reply:
Actually, the newly added test `testListenerRegistrationWhileANewMemberJoining` failed without it with:
```
java.lang.NullPointerException: null
	at com.hazelcast.internal.cluster.impl.operations.OnJoinOp.run(OnJoinOp.java:85) ~[classes/:?]
```